<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>image_processing</class>
 <widget class="QDialog" name="image_processing">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>400</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1200</width>
    <height>700</height>
   </size>
  </property>
  <property name="cursor">
   <cursorShape>ArrowCursor</cursorShape>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::PreventContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>test</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <property name="sizeGripEnabled">
   <bool>false</bool>
  </property>
  <widget class="QPushButton" name="btn1">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>19</y>
     <width>101</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="styleSheet">
    <string notr="true">background: #ffa31a ;color : black;
 border-radius: 10px ;</string>
   </property>
   <property name="text">
    <string>Browse</string>
   </property>
  </widget>
  <widget class="QLabel" name="label2">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>60</y>
     <width>400</width>
     <height>400</height>
    </rect>
   </property>
   <property name="cursor">
    <cursorShape>ArrowCursor</cursorShape>
   </property>
   <property name="styleSheet">
    <string notr="true">border-top-left-radius : 10px;
color: rgb(255, 0, 0);
font: 16pt &quot;MS Shell Dlg 2&quot;;
text-align: center;
background: qlineargradient(spread:reflect, x1:0, y1:0, x2:1, y2:0,stop:0 rgba(116, 200, 255, 255), stop:1 rgba(39, 80, 220, 255));</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="textFormat">
    <enum>Qt::AutoText</enum>
   </property>
   <property name="scaledContents">
    <bool>false</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="btn2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>20</y>
     <width>91</width>
     <height>41</height>
    </rect>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="styleSheet">
    <string notr="true">background: #C0C0C0 ;border-radius: 10px ;
</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="autoDefault">
    <bool>false</bool>
   </property>
   <property name="default">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QComboBox" name="cmb1">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>625</x>
     <y>200</y>
     <width>150</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="editable">
    <bool>false</bool>
   </property>
   <property name="duplicatesEnabled">
    <bool>false</bool>
   </property>
   <item>
    <property name="text">
     <string>Given range</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,1,3,...,119,...,253,255</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,3,7,...,119,...,251,255</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,7,15,..,119,..,247,255</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,15,31,..,111,..,239,255</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,31,63,..,127,..,223,255</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,63,127,255</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,127,255</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>0,255</string>
    </property>
   </item>
  </widget>
  <widget class="QRadioButton" name="rd1">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>50</y>
     <width>160</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>RGB to gray scale</string>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QRadioButton" name="rd2">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>100</y>
     <width>160</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>HSV to gray scale</string>
   </property>
  </widget>
  <widget class="QRadioButton" name="rd3">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>150</y>
     <width>160</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>RGB to HSV</string>
   </property>
  </widget>
  <widget class="QRadioButton" name="rd4">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>200</y>
     <width>160</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Quantize image</string>
   </property>
  </widget>
  <widget class="QRadioButton" name="rd5">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>250</y>
     <width>175</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>gray-level histogram</string>
   </property>
  </widget>
  <widget class="QRadioButton" name="rd6">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>310</y>
     <width>175</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>adjust image dynamic</string>
   </property>
  </widget>
  <widget class="QRadioButton" name="rd7">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>370</y>
     <width>180</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>histogram equalization</string>
   </property>
  </widget>
  <widget class="QRadioButton" name="rd8">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>420</y>
     <width>160</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Remove noise</string>
   </property>
  </widget>
  <widget class="QComboBox" name="cmb2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>645</x>
     <y>420</y>
     <width>110</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <item>
    <property name="text">
     <string>Filter matrix</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>3 * 3</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>5 * 5</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>7 * 7</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>9 * 9</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>11 * 11 </string>
    </property>
   </item>
  </widget>
  <widget class="QPushButton" name="btn3">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>510</y>
     <width>120</width>
     <height>50</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="styleSheet">
    <string notr="true">background: #C0C0C0; border-radius: 10px ;</string>
   </property>
   <property name="text">
    <string>process</string>
   </property>
  </widget>
  <widget class="QLabel" name="label1">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>60</y>
     <width>400</width>
     <height>400</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">border-radius: 10px;
background: qlineargradient(spread:reflect, x1:0, y1:0, x2:1, y2:0, stop:0 rgba(39, 80, 220, 255) stop:1 rgba(116, 200, 255, 255),);</string>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="scaledContents">
    <bool>false</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="btn5">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>790</x>
     <y>510</y>
     <width>120</width>
     <height>50</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <italic>false</italic>
    </font>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="styleSheet">
    <string notr="true">background: #C0C0C0 ; border-radius: 10px ;</string>
   </property>
   <property name="text">
    <string>show in pyplot</string>
   </property>
  </widget>
  <widget class="QPushButton" name="btn4">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>1050</x>
     <y>510</y>
     <width>120</width>
     <height>50</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="styleSheet">
    <string notr="true">background: #C0C0C0 ;border-radius: 10px ;</string>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinbox1">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>320</y>
     <width>60</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="wrapping">
    <bool>true</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
   <property name="maximum">
    <number>255</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinbox2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>320</y>
     <width>60</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="wrapping">
    <bool>true</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
   <property name="maximum">
    <number>255</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="spinbox3">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>380</y>
     <width>60</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="wrapping">
    <bool>true</bool>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
   <property name="maximum">
    <number>255</number>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>630</x>
     <y>290</y>
     <width>60</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>min range</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>290</y>
     <width>60</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>max range</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>350</y>
     <width>60</width>
     <height>30</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>max value</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
